OrbitGl/GlSlider.cpp

```diff
@@ -34,41 +32,76 @@ void GlSlider::SetSliderWidthRatio(float a_WidthRatio)  // [0,1]
   m_Length = std::max(a_WidthRatio, minWidth);
 }
 
-void GlSlider::OnPick(int a_X, int a_Y) {
-  m_Vertical ? OnPickVertical(a_X, a_Y) : OnPickHorizontal(a_X, a_Y);
+Color GlSlider::GetLighterColor(const Color& color) {
+  return Color(static_cast<unsigned char>(color[0] * 1.25),
+               static_cast<unsigned char>(color[1] * 1.25),
+               static_cast<unsigned char>(color[2] * 1.25), 255);
 }
 
-void GlSlider::OnPickHorizontal(int a_X, int /*a_Y*/) {
-  float canvasWidth = m_Canvas->getWidth();
-  float sliderWidth = m_Length * canvasWidth;
-  float nonSliderWidth = canvasWidth - sliderWidth;
+Color GlSlider::GetDarkerColor(const Color& color) {
+  return Color(static_cast<unsigned char>(color[0] * 0.75),
+               static_cast<unsigned char>(color[1] * 0.75),
+               static_cast<unsigned char>(color[2] * 0.75), 255);
+}
 
-  float x = static_cast<float>(a_X);
-  float sliderX = m_Ratio * nonSliderWidth;
-  m_PickingRatio = (x - sliderX) / sliderWidth;
+void GlSlider::DrawBackground(GlCanvas* canvas, float x, float y, float width, float height) {
+  Batcher* batcher = canvas->GetBatcher();
+  const Color dark_border_color = GetDarkerColor(m_BarColor);
+
+  // Dark border
+  {
+    Box box(Vec2(x, y), Vec2(width, height), GlCanvas::kZValueSliderBg - 0.0001f);
+    batcher->AddBox(box, dark_border_color, shared_from_this());
+  }
+  // Background itself
+  {
+    Box box(Vec2(x + 1, y + 1), Vec2(width - 2, height - 2), GlCanvas::kZValueSliderBg);
```

Comment:
Vec2 contain floats, we should use 1.f and 2.f, and declare those as constants.

Reply:
Done